### PR TITLE
search redesign: quicklinks in sidebar

### LINKED ISSUE #20330: Search redesign: Implement sidebar quicklinks section

### PR DESCRIPTION
**Pull request by limitedmage**
Fixes #20330

- Adds quicklinks to sidebar
- Hides empty sidebar sections

image

### CODE REVIEW

File: client/web/src/search/results/streaming/sidebar/SearchTypeLink.tsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ import { SearchType } from '../../SearchResults'
 import { SearchSidebarProps } from './SearchSidebar'
 import styles from './SearchSidebarSection.module.scss'
 
-interface SearchTypeLinkProps extends SearchSidebarProps {
+interface SearchTypeLinkProps extends Omit<SearchSidebarProps, 'settingsCascate'> {
```
Comment (rvantonder) [inline client/web/src/search/results/streaming/sidebar/SearchTypeLink.tsx:12]:
```suggestion
interface SearchTypeLinkProps extends Omit<SearchSidebarProps, 'settingsCascade'> {
```

File: client/web/src/search/results/streaming/sidebar/SearchTypeLink.tsx [hunk 2]
```diff
@@ -40,7 +40,7 @@ const SearchTypeLink: React.FunctionComponent<SearchTypeLinkProps> = ({
     )
 }
 
-export const getSearchTypeLinks = (props: SearchSidebarProps): ReactElement[] => {
+export const getSearchTypeLinks = (props: Omit<SearchSidebarProps, 'settingsCascate'>): ReactElement[] => {
```
Comment (rvantonder) [inline client/web/src/search/results/streaming/sidebar/SearchTypeLink.tsx:43]:
```suggestion
export const getSearchTypeLinks = (props: Omit<SearchSidebarProps, 'settingsCascade'>): ReactElement[] => {
```

Comment (limitedmage) [reply client/web/src/search/results/streaming/sidebar/SearchTypeLink.tsx:43]: 🤦‍♀️

Comment (rvantonder) [reply client/web/src/search/results/streaming/sidebar/SearchTypeLink.tsx:43]: facepalm

**Review by rvantonder — APPROVED** [review]
Just gonna stamp all these.

**Pull request merged**